packages/atomic/src/components/commerce/atomic-commerce-interface/atomic-commerce-interface.tsx

```diff
@@ -0,0 +1,480 @@
+    return this.internalInitialization(() => (this.engine = engine));
+  }
+
+  /**
+   *
+   * Executes the first search and logs the interface load event to analytics, after initializing connection to the headless search engine.
+   */
+  @Method() public async executeFirstSearch() {
+    if (!this.commonInterfaceHelper.engineIsCreated(this.engine)) {
+      return;
+    }
+
+    if (!this.initialized) {
+      console.error(
+        'You have to wait until the "initialize" promise is fulfilled before executing a search.',
+        this.host
+      );
+      return;
+    }
+
+    if (this.localizationCompatibilityVersion !== 'v4') {
+      this.engine.logger.warn(
+        `As of Atomic version 3.0.0, support for JSON compatibility ${this.localizationCompatibilityVersion} will be deprecated. Please update the JSON compatibility to v4: <atomic-commerce-interface localization-compatibility-version="v4" ...></atomic-commerce-interface> For more information, see i18next Migration Guide: https://www.i18next.com/misc/migration-guide#v20.x.x-to-v21.0.0.`
+      );
+    }
+
+    const safeStorage = new SafeStorage();
+
+    const standaloneSearchBoxData =
+      safeStorage.getParsedJSON<StandaloneSearchBoxData | null>(
+        StorageItems.STANDALONE_SEARCH_BOX_DATA,
+        null
+      );
+
+    if (!standaloneSearchBoxData) {
+      //this.engine.executeFirstSearch();
+      return;
+    }
+
+    safeStorage.removeItem(StorageItems.STANDALONE_SEARCH_BOX_DATA);
+    //const {updateQuery} = loadQueryActions(this.engine!);
+    const {value} = standaloneSearchBoxData;
+    this.engine!.dispatch(updateQuery({query: value}));
+    //  this.engine.executeFirstSearchAfterStandaloneSearchBoxRedirect();
+  }
+
+  /**
+   * Returns the unique, organization-specific endpoint(s).
+   * @param {string} organizationId
+   * @param {'prod'|'hipaa'|'staging'|'dev'} [env=Prod]
+   */
+  @Method() public async getOrganizationEndpoints(
+    organizationId: string,
+    env: PlatformEnvironment = 'prod'
+  ) {
+    return getOrganizationEndpointsHeadless(organizationId, env);
+  }
+
+  public get bindings(): CommerceBindings {
+    return {
+      engine: this.engine!,
+      i18n: this.i18n,
+      store: this.store,
+      interfaceElement: this.host,
+      createStyleElement: () => {
+        const styleTag = document.createElement('style');
+        if (this.CspNonce) {
+          styleTag.setAttribute('nonce', this.CspNonce);
+        }
+        return styleTag;
+      },
+      createScriptElement: () => {
+        const styleTag = document.createElement('script');
+        if (this.CspNonce) {
+          styleTag.setAttribute('nonce', this.CspNonce);
+        }
+        return styleTag;
+      },
+    };
+  }
+
+  private initFieldsToInclude() {
+    const fields = EcommerceDefaultFieldsToInclude.concat(this.fieldsToInclude);
+    this.store.addFieldsToInclude(fields);
+  }
+
+  public registerFieldsToInclude() {
```

Comment:
Related to above comment: Fields to include support might not be necessary

Reply:
Thats required in BaseAtomicInterface. Do I modify BaseAtomicInterface so that registerFieldsToInclude is optional, or do I avoid using the interface?

Reply:
I would modify `BaseAtomicInterface`, yes !